src/ui/Card.tsx

```diff
@@ -58,22 +62,28 @@ export function Card(props: CardProps): JSX.Element {
   // parallel with swiping the card offscreen.
   const [hasSwiped, setHasSwiped] = useState(false);
 
-  async function swipeLeft(): Promise<void> {
+  async function handleSwipe(
+    addLabelIds: string[],
+    removeLabelIds: string[],
+  ): Promise<void> {
     if (hasSwiped) {
       return;
     }
     setHasSwiped(true);
     assert(messages.length);
-    await props.actions.archive(messages);
+    await modifyMessages(messages, addLabelIds, removeLabelIds);
+  }
+
```

Comment:
Should Inbox have a LabelName?

Is it worth it to lazily initialize labels? Or should we just getOrCreate all of them up front?

Reply:
Oh, giving inbox a LabelName makes sense. Reduces possibility of typos and less special-caseing of code.

I'm torn about the lazy initialize labels thing. It would certainly make the code nicer to not need that extra async call in there and there are places where we read labels that could then just be sync. But then the initial page load needs at least the round-trip to fetch all labels and the first time you load teamail it needs more round trips to create all the labels. Also, if you don't use some of these labels it still clutters your gmail with them. Although we could fix that by only creating the labels you actually have actions for.

I dunno. WDYT?

Reply:
Could we just improve the ergonomics?

Something like 
await handleSwipe([await labelIdFor(LabelName.keep), await labelIdFor(LabelName.emptyDaily)]

might not be too bad?

Then labelIdFor does the lazy initialization behind the scenes?

Reply:
That sounds fine to me. There will likely be calling locations that want the whole Label instance, but we can have both if needed.